styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -159,17 +173,44 @@ void tick() {
 
     timedOutInstances.forEach(this::sendTimeout);
 
-    for (InstanceState eligibleInstance : eligibleInstances) {
-      final boolean proceed = limitAndDequeue(
-          resources, workflowResourceReferences, currentResourceUsage, eligibleInstance);
-      if (!proceed) {
-        break;
-      }
-    }
+    limitAndDequeue(config, resources, workflowResourceReferences, currentResourceUsage,
+        eligibleInstances);
 
     updateStats(resources, currentResourceUsage);
   }
 
+  private void limitAndDequeue(
+      StyxConfig config,
```

Comment:
It doesn't feel very necessary to pass in the whole `StyxConfig` object, a boolean would just work.

Reply:
Doesn't feel very necessary to pull out an individual field either... =)

Reply:
This was mostly a nitpicking, so 👍